src/lib.rs

```diff
@@ -1048,4 +1051,79 @@ mod test {
             .unwrap();
         stream.shutdown().unwrap();
     }
+
+    #[test]
+    fn howsmyssl() {
+        #[derive(RustcDecodable, PartialEq, Eq)]
+        struct HowsMySslResults {
+            given_cipher_suites: Vec<String>,
+            ephemeral_keys_supported: bool,
+            session_ticket_supported: bool,
+            tls_compression_supported: bool,
+            unknown_cipher_suite_supported: bool,
+            beast_vuln: bool,
+            able_to_detect_n_minus_one_splitting: bool,
+            insecure_cipher_suites: HashMap<String, Vec<String>>,
+            tls_version: String,
+            rating: String,
+        }
+
+        let creds = SchannelCredBuilder::new().acquire(Direction::Outbound).unwrap();
+        let stream = TcpStream::connect("www.howsmyssl.com:443").unwrap();
+        let mut stream = TlsStreamBuilder::new()
+            .domain("www.howsmyssl.com")
+            .initialize(creds, stream)
+            .unwrap();
+        stream.write_all(b"GET /a/check HTTP/1.0\r\nHost: www.howsmyssl.com\r\n\r\n").unwrap();
+        let mut out = vec![];
+        stream.read_to_end(&mut out).unwrap();
+        assert!(out.starts_with(b"HTTP/1.0 200 OK"));
+
+        // Retrieve the JSON
+        let mut index = 0;
+        for _ in out.windows(4).take_while(|s| s != &[b'\r', b'\n', b'\r', b'\n']) {
+            index += 1;
+        }
+        let json = String::from_utf8(out[index + 4..].to_owned()).unwrap();
+
+        // Compare the results
+        let results: HowsMySslResults = json::decode(&json).unwrap();
+        assert_eq!(results.given_cipher_suites, vec![
```

Comment:
It seems like this list will vary across things like OS versions - maybe just check that a couple of known-universal suites are present?